ui/apps/dashboard/src/app/(organization-active)/(dashboard)/integrations/vercel/callback/connect.tsx

```diff
@@ -0,0 +1,171 @@
+'use client';
+
+import { useState } from 'react';
+import { type Route } from 'next';
+import { useRouter } from 'next/navigation';
+import { NewButton } from '@inngest/components/Button/index';
+import { Card } from '@inngest/components/Card/Card';
+import { Checkbox } from '@inngest/components/Checkbox/Checkbox';
+import {
+  RiArrowLeftSLine,
+  RiArrowRightSLine,
+  RiCloseLine,
+  RiInformationLine,
+} from '@remixicon/react';
+
+import Input from '@/components/Forms/Input';
+import type VercelIntegration from '../../../settings/integrations/vercel/VercelIntegration';
+import useUpdateVercelIntegration from '../../../settings/integrations/vercel/useUpdateVercelIntegration';
+import type { VercelCallbackProps } from './page';
+
+const PAGE_SIZE = 4;
+
+export default function Connect({
+  searchParams,
+  integrations,
+}: VercelCallbackProps & { integrations: VercelIntegration }) {
+  const router = useRouter();
+  const [projects, setProjects] = useState(integrations.projects);
+  const [saving, setSaving] = useState(false);
+  const [page, setPage] = useState(1);
+  const start = (page - 1) * PAGE_SIZE;
+  const [hover, setHover] = useState(null);
+  const [custom, setCustom] = useState<string[]>([]);
+
+  const updateVercelIntegration = useUpdateVercelIntegration(integrations);
+
+  const check = (id: string, enabled: boolean) =>
+    setProjects(projects.map((p) => ({ ...p, isEnabled: p.id === id ? enabled : p.isEnabled })));
+
+  const setPath = (id: string, value: string) =>
+    setProjects(projects.map((p) => ({ ...p, servePath: p.id === id ? value : p.servePath })));
+
+  const pages = Math.ceil(projects.length / PAGE_SIZE);
+  const end = Math.min(start + PAGE_SIZE, projects.length);
+
+  const submit = async () => {
+    await updateVercelIntegration({
+      ...integrations,
+      projects,
+    });
+    setSaving(false);
+    router.push(
+      `/integrations/vercel/callback/success?onSuccessRedirectURL=${searchParams.next}` as Route
+    );
+  };
+
+  return (
+    <>
+      <Card className="w-full border-slate-200">
```

Comment:
todo (next, on a separate PR): use the color tokens everywhere. In the case of some components (like Card), they already have color tokens, so no need for extra styles at all

Reply:
I'm about half way through this work and will continue. Will push the rest as a follow up if we are ok with everything else so far.